Bottom: f1083d042db3658ee98dca601a0a2fd94d4cb12a
Top:    3fd71bbdde8064c89162f5d4672b6b5bead9703a
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-20 09:12:46 -0700

Refresh of rs-fix

---

diff --git a/src/preload.c b/src/preload.c
index 8f19af5..474287e 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -520,7 +520,7 @@ static void fork_active(int socket)
 	if ((ret != sizeof msg) || msg)
 		goto err2;
 
-	ret = rconnect(ret, (struct sockaddr *) &addr, len);
+	ret = rconnect(dfd, (struct sockaddr *) &addr, len);
 	if (ret)
 		goto err2;
 
@@ -618,9 +618,9 @@ static inline enum fd_type fd_fork_get(int index, int *fd)
 
 	fdi = idm_lookup(&idm, index);
 	if (fdi) {
-		if (fdi->type == fd_fork_passive)
+		if (fdi->state == fd_fork_passive)
 			fork_passive(index);
-		else if (fdi->type == fd_fork_active)
+		else if (fdi->state == fd_fork_active)
 			fork_active(index);
 		*fd = fdi->fd;
 		return fdi->type;
